src/components/data/DataTable.js

```diff
@@ -281,16 +284,23 @@ const DataTable = (props) => {
     return entities.length && _.every((k) => _.includes(k, selectedKeys), entityKeys);
   };
 
-  const searchByColumn = (field, v) => {
+  const searchByColumn = (field, v, type) => {
     setActiveTextFilter('');
     setColumnFilter({ filterColAttr: field, filterColTerm: v.toString().trim() });
     setPageNumber(1);
-    Ajax().Metrics.captureEvent(Events.workspaceDataFilteredSearch, {
-      workspaceNamespace: namespace,
-      workspaceName: name,
-      providerName: dataProvider.providerName,
-      cloudPlatform: dataProvider.providerName === wdsProviderName ? cloudProviders.azure.label : cloudProviders.gcp.label,
-    });
+    if (type === 'name') {
+      Ajax().Metrics.captureEvent(Events.workspaceDataFilteredSearch, {
+        ...extractWorkspaceDetails(workspace.workspace),
+        searchType: 'filter-by-name',
+        providerName: dataProvider.providerName,
+      });
+    } else {
+      Ajax().Metrics.captureEvent(Events.workspaceDataFilteredSearch, {
+        ...extractWorkspaceDetails(workspace.workspace),
+        searchType: 'filter-by-column',
+        providerName: dataProvider.providerName,
+      });
+    }
```

Comment:
This could use the value of `field` to determine the `searchType` instead of adding another argument.

Also, instead of two different values for `searchType`, would it be preferable to label both these cases as `filter-by-column` and have another flag like `isIdColumn`?

```suggestion
    Ajax().Metrics.captureEvent(Events.workspaceDataFilteredSearch, {
      ...extractWorkspaceDetails(workspace.workspace),
      searchType: 'filter-by-column',
      isIdColumn: field === entityMetadata[entityType].idName,
      providerName: dataProvider.providerName,
    });
```

Or keeping the two different search types, this could still be condensed a little.
```suggestion
    Ajax().Metrics.captureEvent(Events.workspaceDataFilteredSearch, {
      ...extractWorkspaceDetails(workspace.workspace),
      searchType: type === 'name' ? 'filter-by-name' : 'filter-by-column',
      providerName: dataProvider.providerName,
    });
```

Reply:
Thank you so much for the suggestion. The ticket suggested 3 separate cases so I think the latter is the best option.